### DISCUSSION TITLE
HasMany, primaryKey

### DISCUSSION THREAD

**Discussion by codexJoin** [sequelize/sequelize#15696] [Help & Questions] [answered]
Hello, i have two tables called "Users" and "Controls".
The "Controls" table cannot have a key called id with autoincrement.
Because a user can have multiple controls.
It turns out that the system needs the Primary key to be "userID",
that is, has many returns only the last record of Controls.
The right thing would be to bring everyone.

For example userId 1043 have 5 controls, the table looks like this:
userID | codcontrol:
1043 10
1043 20
1043 30
1043 45
1043 12
But the hasMany return only the last.
I cannot insert a new field called "id", as the database is still used by an old system.
When i insert an field called "id" it's working normal.

Structure table "Users" :
```js
'use strict';
module.exports = (sequelize, DataTypes) => {
  const User = sequelize.define('Users', {
    email: DataTypes.STRING,
  }, {});

  User.associate = models => {
    User.hasMany(models.Contros, {
      onDelete: "cascade",
      foreignKey: 'userID',
    });
  }
  return User;
};
```

Structure table "Controls":
```js
module.exports = (sequelize, DataTypes) => {
    const Controls = sequelize.define("Controls", {
        userID: {
            type: DataTypes.INTEGER,
            unique: false,
            autoIncrement: false,
            primaryKey: true,
            allowNull: false,
            /*references: {
                model: 'Users',
                key: 'id'
            }*/
        },
        codecontrol: {
            type: DataTypes.INTEGER,
            otherKey: true,
        }
    },{
        timestamps: false
    });

    Controls.associate = models => {
        Controls.belongsTo(models.User, {
            foreignKey: 'userID',
            targetKey: 'userID',
        });
    }

    return Controls;
}
```

**Comment by ephys** [comment, marked as answer]
If you need one user to have multiple Controls, you should not use the primary key of Controls as the foreign keys referencing User, as primary keys are unique

**Reply by codexJoin** [reply]
I already solved the problem, but thanks for the answer

**Reply by ephys** [reply]
You should consider updating threads you open with the solution you used, as it can be useful to other people encountering the same issue

**Reply by codexJoin** [reply]
I just specified the foreignKey and sourceKey correctly and it worked.
